tests/ctst/common/hooks.ts

```diff
@@ -37,8 +49,129 @@ Before({ tags: '@Quotas', timeout: 1200000 }, async function (scenarioOptions) {
     await prepareQuotaScenarios(this as Zenko, scenarioOptions);
 });
 
+After(async function (this: Zenko, results) {
+    if (results.result?.status === 'FAILED') {
+        return;
+    }
+    await cleanS3Bucket(
+        this,
+        this.getSaved<string>('bucketName'),
+    );
+});
+
 After({ tags: '@Quotas' }, async function () {
     await teardownQuotaScenarios(this as Zenko);
 });
 
+After(async function (this: Zenko, results) {
```

Comment:
this seems mostly related to the iam tests: and we actually don't do anything if there is no saved `crossAccountName`...
- similar to `teardownQuotaScenarios`, would be better to move this cleanup logic to the associated test suite (even if we need to keep the `After` here) : to have better "locality" of the test & associated cleanup code
- since it is specific to these tests, maybe we should use a "tag filter" (like what is done for quotas) to better express this should happen only for these specific tests? (instead of just relying on `crossAccountName`). It would be more explicit.

Reply:
-> ZENKO-4904